src/main.rs

```diff
@@ -108,12 +108,19 @@ async fn main() -> anyhow::Result<()> {
             )
             .await;
 
-            download_images(&images, &client, &download_path)
+            let images_to_process = get_images_info_to_process(
+                &images,
+                &download_path,
+                cli.download_only_unexisting_in_folder,
+            ).await;
+
+            download_images(&images_to_process, &client)
                 .await
                 .expect("Error downloading");
             // optimizations doesn't seem to work with threads/futures
             if !cli.opt_only_on_validation {
-                for img in images {
+                for img_info in images_to_process {
+                    let img = img_info.0;
                     let path = if img.scale == 1 {
```

Comment:
I think we can also get rid of this as you have previously found the path for each image. It would be safe to just use `img.1`.

Reply:
good catch!

Reply:
done!